pyo3-ffi/build.rs

```diff
@@ -56,15 +56,21 @@ fn ensure_python_version(interpreter_config: &InterpreterConfig) -> Result<()> {
                 interpreter_config.version,
                 versions.min,
             );
-            ensure!(
-                interpreter_config.version <= versions.max || env_var("PYO3_USE_ABI3_FORWARD_COMPATIBILITY").map_or(false, |os_str| os_str == "1"),
-                "the configured Python interpreter version ({}) is newer than PyO3's maximum supported version ({})\n\
-                 = help: please check if an updated version of PyO3 is available. Current version: {}\n\
-                 = help: set PYO3_USE_ABI3_FORWARD_COMPATIBILITY=1 to suppress this check and build anyway using the stable ABI",
-                interpreter_config.version,
-                versions.max,
-                std::env::var("CARGO_PKG_VERSION").unwrap(),
-            );
+            if !interpreter_config
+                .build_flags
+                .0
+                .contains(&BuildFlag::Py_GIL_DISABLED)
+            {
+                ensure!(
+                        interpreter_config.version <= versions.max || env_var("PYO3_USE_ABI3_FORWARD_COMPATIBILITY").map_or(false, |os_str| os_str == "1"),
```

Comment:
This transformation disables the version check completely on the free-threaded build; I think this is risky given testing with 3.14 free-threading is likely to start happening soon.

I think better would be to do something like

```rust
if interpreter_config.version < versions.max {
    if interpreter_config.build_flags.0.contains(&BuildFlags::Py_GIL_DISABLED) {
        // free-threaded specific error
    } else if !env_var("PYO3_USE_ABI3_FORWARD_COMPATIBILITY").map_or(false, |os_str| os_str == "1") {
        // env var not set, raise the existing error
    }
}
```

Reply:
I did this and noticed we were using `interpreter_config.build_flags.0.contains(&BuildFlags::Py_GIL_DISABLED)` quite a bit, so I added that as a method to the `InterpreterConfig` struct to make the spelling a little clearer